internal/test/framework/framework.go

```diff
@@ -40,35 +45,53 @@ type TestFramework struct {
 	SSHClient *ssh.Client
 	// CloudProvider holds the information related to cloud provider
 	cloudProvider cloudprovider.Cloud
+}
+
+// TestFramework holds the info to run the test suite.
+// This is not clean
+type TestFramework struct {
+	RemoteDir string
+	WinVMs    []windowsVM
 	// k8sclientset is the kubernetes clientset we will use to query the cluster's status
 	K8sclientset *kubernetes.Clientset
 	// OSConfigClient is the OpenShift config client, we will use to query the OpenShift api object status
 	OSConfigClient *configclient.Clientset
 }
 
-// Setup sets up the Windows node so that it can join the existing OpenShift cluster
-// TODO: move this to return error and do assertions there
-func (f *TestFramework) Setup() {
-	if err := f.createWindowsVM(); err != nil {
-		log.Fatalf("failed to create Windows VM: %v", err)
+func initCIvars() error {
+	kubeconfig = os.Getenv("KUBECONFIG")
+	if kubeconfig == "" {
+		return fmt.Errorf("KUBECONFIG environment variable not set")
 	}
-	// TODO: Add some options to skip certain parts of the test
-	if err := f.setupWinRMClient(); err != nil {
-		log.Fatalf("failed to setup winRM client for the Windows VM: %v", err)
+	awsCredentials = os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
+	if awsCredentials == "" {
+		return fmt.Errorf("AWS_SHARED_CREDENTIALS_FILE environment variable not set")
 	}
-	// Wait for some time before starting configuring of ssh server. This is to let sshd service be available
-	// in the list of services
-	// TODO: Parse the output of the `Get-Service sshd, ssh-agent` on the Windows node to check if the windows nodes
-	// has those services present
-	time.Sleep(time.Minute)
-	if err := f.configureOpenSSHServer(); err != nil {
-		log.Fatalf("failed to configure OpenSSHServer on the Windows VM: %v", err)
+	artifactDir = os.Getenv("ARTIFACT_DIR")
+	if awsCredentials == "" {
+		return fmt.Errorf("ARTIFACT_DIR environment variable not set")
+	}
+	privateKeyPath = os.Getenv("KUBE_SSH_KEY_PATH")
+	if privateKeyPath == "" {
+		return fmt.Errorf("KUBE_SSH_KEY_PATH environment variable not set")
 	}
-	if err := f.createRemoteDir(); err != nil {
-		log.Fatalf("failed to create remote dir with error: %v", err)
+	return nil
+}
+
+// Setup sets up the Windows node so that it can join the existing OpenShift cluster
+// TODO: move this to return error and do assertions there
+func (f *TestFramework) Setup(nrVMs int) {
+	if err := initCIvars(); err != nil {
+		log.Fatalf("failed to initialize CI variables with error: %v", err)
 	}
-	if err := f.getSSHClient(); err != nil {
-		log.Fatalf("failed to get ssh client for the Windows VM created: %v", err)
+
+	f.WinVMs = make([]windowsVM, nrVMs)
+	// TODO: make them run in parallel
```

Comment:
Now that JIRA is public we have the option of linking tickets in comments. I'd think it would be nice for TODO's to each have a ticket associated with them, so planned work isn't lost. @aravindhp thoughts on that?